azure-active-directory-server/src/main/java/org/jetbrains/teamcity/aad/AADAuthenticationScheme.java

```diff
@@ -121,7 +122,12 @@ public HttpAuthenticationResult processAuthenticationRequest(@NotNull HttpServle
     final String email = token.getClaim(EMAIL_CLAIM);
 
     final ServerPrincipal principal = myPrincipalFactory.getServerPrincipal(name, oid, email, schemeProperties);
-
+   
+    if(principal == null)
```

Comment:
@ekoshkin I see. That behavior will be useful, but I'm not getting any message, if the _new user creation_ is disabled, the login form is reloaded without any message. I'm on 10.0.2 (build 42234).